### ISSUE TITLE
Intuitive simple slider mode

### ISSUE THREAD

**Issue by MacroMachines** [tooll3/t3#492]
I am absolutely loving the new version of Toolll3, 

one of the main pain points I am experiencing thus far in my relatively early usage of it is that I tried all of the value input method settings and I find all of them very difficult to get to a value I want efficiently and often end up having to type in the value.  The default infinity slider I suspect may be a point of confusion for others as well as it functions very differently than any slider widget that I have encountered and requires thinking in terms of scale magnification on one axis and the value adjustment on the other which is nonlinear, and if I move my mouse up or down a tiny bit unintentionally my understanding of where my value should be moving falls apart.  and the rotary options all require challenging mouse movements.

Notch has a very intuitive and simple yet capable slider design for rapidly adjusting values in a precise yet intuitive way, here is a video of me showing how it works: https://www.dropbox.com/scl/fi/p4fo7fr92fe5bj1mjqjof/2024-12-02-19-21-31.mp4?rlkey=30q3izzo84xcuajmsxlf2k2vx&dl=0

The slider has a visual horizontal slider with the width of the parameters value box, parameters have a default range that is set so the common range of values that one would typically use for that parameter equates to the visual left and right sides of the slider, like a simple common horizontal linear slider widget, BUT if you hold control and drag you can drag past the edges and get to any value, and when you release the click, that now becomes the minimum or maximum range for that widget, so you can intuitively adjust settings rapidly but still get to any value you could want, and if you click it to enter the text value and that is higher or lower than the current min and max range it will also scale the slider range.

This also helps a TON with working on laptops or touch screens as well as in live performance.  Having a value range where you know what is going to happen if you move it to the left or right sides is more performance friendly, akin to using midi inputs, great for touch screen live control.

**Comment by MacroMachines** [comment]
Another reinforcement as to why I belive this to be an important feature request is that I've crashed Tooll3 a few times now by moving the slider to a value range that jumped dramatically from where I was trying to go.

**Comment by MacroMachines** [comment]
I think another aspect of why it becomes increasingly challenging to use the current widgets available is that when the graphics card gets overwhelmed the GUI response goes down a bit and then adjusting gets extra difficult and the mouse can jump further than expected, which gets exponentially challenging when the values go crazy high or low as they can begin to stress the GPU further and then it all spirals out of control.

**Comment by pixtur** [comment]
Excellent request, especially with the video capture! I can really see the use case for tweaking input ranges.

Some points here:
- Note that blue values are connected (driven by) an input and can't be edited.
- You can edit the value range of parameters by right clicking on the parameter name and selecting "Parameter Settings".
- Sadly the value-range is defined for the Symbol, and will effect all instances of it. So for changing that to the notch-behaviour, this would need to change.
- There actually IS a virtual slider behind the numbers. Try editing the channels of [AColor].
- The infinity-slider needs some getting used to, but is definitely not everybody's cup of tea. You can change the gizmo in the settings window.

In the notch example, what happens if the manually entered value falls within the current value range? Let's say the slider is defined for the 0 ... 10 and you enter 5? Is this now the upper (or lower) new boundary?

**Comment by dbxnls** [comment]
Hello ! I must say tooll is really interesting and i am still pretty new using it. My background is from the visual realm using vdmx,touchdesigner,notch,mm etc. First of all i am on an old lowend computer running an 970 nvidia gpu and one of 
my issue is actually the infinity slider interface. If i by mistake push the infinity slider wrong the values make my computer totally choke and i have to force quit. This happens allot. Thank you @MacroMachines  for adding this post  (issue). I love tooll and i wish i could be friend with the infinity slider but it's really though. I am used to live patch shows when using vdmx and it works well. One solution im looking into now is to actually make an touchosc interface to lock down ranges but then the live "patching" thing would get limited.  Adding a link below how sliders and inspectors is solved in vdmx.

https://vdmx.vidvox.net/tutorials/the-slider-ui-item-in-depth

**Comment by MacroMachines** [comment]
Yeah, maybe this should become a properly filed bug with steps to reproduce somehow?:

> when the graphics card gets overwhelmed, the GUI response goes down a bit  
adjusting parameters with these resolution scales when the mouse is jumping further than expected becomes extra difficult, 
the values go extremely high or low and begin to stress the GPU further, 
then the system will hang and freeze the application.

**Comment by pixtur** [comment]
Hi @dbxnls thanks for your feedback. Have you tried switching the infinity-slider to one of the other input methods:

https://github.com/user-attachments/assets/be347df6-ffd9-4467-97d5-0f08895b87ef

**Comment by pixtur** [comment]
Closing this for now.
